### PR TITLE
MINIFICPP-2048 ARM64 support part 1

### PR DESCRIPTION
**Pull request by martinzink**
MINIFICPP-2049 update crc32c
MINIFICPP-2050 AWS tests compilation fix for ARM
MINIFICPP-2051 updated ossp-uuid config.guess
MINIFICPP-2052 unsigned/signed char fixes

With these changes I was able to compile the project on aarch64 rockylinux 9 with the default extensions.

---
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [X] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [X] If applicable, have you updated the LICENSE file?
- [X] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/test/unit/StringUtilsTests.cpp [hunk 1]
```diff
@@ -388,30 +390,22 @@ TEST_CASE("TestStringUtils::testBase64Decode", "[test base64 decode]") {
...
+      static_cast<char>(-117), static_cast<char>(48), static_cast<char>(-45), static_cast<char>(-113),
+      static_cast<char>(65), static_cast<char>(20), static_cast<char>(-109), static_cast<char>(81),
+      static_cast<char>(85), static_cast<char>(-105), static_cast<char>(97), static_cast<char>(-106),
+      static_cast<char>(-101), static_cast<char>(113), static_cast<char>(-41), static_cast<char>(-97),
+      static_cast<char>(-126), static_cast<char>(24), static_cast<char>(-93), static_cast<char>(-110),
+      static_cast<char>(89), static_cast<char>(-89), static_cast<char>(-94), static_cast<char>(-102),
+      static_cast<char>(-85), static_cast<char>(-78), static_cast<char>(-37), static_cast<char>(-81),
+      static_cast<char>(-61), static_cast<char>(28), static_cast<char>(-77), static_cast<char>(-45),
+      static_cast<char>(93), static_cast<char>(-73), static_cast<char>(-29), static_cast<char>(-98),
+      static_cast<char>(-69), static_cast<char>(-13), static_cast<char>(-33), static_cast<char>(-65)};
```
Comment (szaszm) [inline libminifi/test/unit/StringUtilsTests.cpp:405]: Can we represent the same in a more compact way, like this?
```suggestion
  std::string expected = 
    "\x00\x10\x83\x10"
    "\x51\x87\x20\x92"
    "\x8b\x30\xd3\x8f"
    "\x41\x14\x93\x51"
    "\x55\x97\x61\x96"
    "\x9b\x71\xd7\x9f"
    "\x82\x18\xa3\x92"
    "\x59\xa7\xa2\x9a"
    "\xab\xb2\xdb\xaf"
    "\xc3\x1c\xb3\xd3"
    "\x5d\xb7\xe3\x9e"
    "\xbb\xf3\xdf\xbf";
```

File: thirdparty/ossp-uuid/ossp-uuid-update-config-guess.patch [hunk 2]
```diff
@@ -0,0 +1,5538 @@
+    diff --git a/config.guess b/config.guess
+index ca2a03c..69188da 100755
+--- a/config.guess
++++ b/config.guess
```
Comment (szaszm) [inline thirdparty/ossp-uuid/ossp-uuid-update-config-guess.patch:4]: Where is this huge patch coming from?

Comment (martinzink) [reply thirdparty/ossp-uuid/ossp-uuid-update-config-guess.patch:4]: They are the newer version of the included config.guess and config.sub
https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess
https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub

File: libminifi/test/unit/StringUtilsTests.cpp (hunk 1 above)
Comment (martinzink) [reply libminifi/test/unit/StringUtilsTests.cpp]: Good idea, unfortunetly this version fails for some reason on my machine. But I've created [something similar](https://github.com/apache/nifi-minifi-cpp/pull/1517/commits/8526c237224d7f9a67d88bc0fca6428e1e873083) based on your idea.

File: thirdparty/ossp-uuid/ossp-uuid-update-config-guess.patch (hunk 2 above)
Comment (szaszm) [reply thirdparty/ossp-uuid/ossp-uuid-update-config-guess.patch:4]: Can we update the whole package instead?

**Review by szaszm — APPROVED** [review]

File: libminifi/test/unit/StringUtilsTests.cpp (hunk 1 above)
Comment (adamdebreceni) [reply libminifi/test/unit/StringUtilsTests.cpp]: it probably fails due to the leading null terminator `\x00`

**Review by adamdebreceni — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

**Pull request closed without merging**